### PR TITLE
feat: add convenience function to wallet for easy withdrawal to base chain

### PR DESCRIPTION
**Pull request by pixelcircuits**
Adds a convenience function for withdraw base asset to the base chain. This will be useful for generating test output message for future issues to test with

### CODE REVIEW

File: packages/wallet/src/transfer.test.ts [hunk 1]
```diff
@@ -114,4 +116,25 @@ describe('Wallet', () => {
...
+    const tx = await sender.withdraw(recipient, 10);
+    const result = await tx.wait();
+
+    const messageOutReceipt = <TransactionResultMessageOutReceipt>result.receipts[0];
+    expect(result.transactionId).toEqual(messageOutReceipt.sender);
+    expect(recipient.toHexString()).toEqual(messageOutReceipt.recipient);
+    expect(amount.toString()).toEqual(messageOutReceipt.amount.toString());
+
+    const senderBalances = await sender.getBalances();
+    expect(senderBalances).toEqual([{ assetId: NativeAssetId, amount: bn(90) }]);
```
[the remark this answers is not in the document]
Comment (pixelcircuits) [reply packages/wallet/src/transfer.test.ts:138]: The 'recipient' is an Ethereum address. Their balance won't be updated on Ethereum until `relayMessageFromFuelBlock` on the Ethereum side MessagePortal contract is called. Full end-to-end testing of this is currently being implemented in [this repo](https://github.com/FuelLabs/bridge-integration-tests)

**Review by digorithm — CHANGES_REQUESTED** [review]
Looks really nice. What do you think about renaming it from `withdraw()` to `transfer()` so we can keep the terminology/API between the Rust SDK and the TS SDK similar? [The equivalent of `withdraw()` in the Rust SDK is `transfer()`.](https://fuellabs.github.io/fuels-rs/v0.27.0/wallets/transferring-assets.html?highlight=transfer#transferring-assets)

**Comment by digorithm** [comment]
> > Looks really nice. What do you think about renaming it from `withdraw()` to `transfer()` so we can keep the terminology/API between the Rust SDK and the TS SDK similar? [The equivalent of `withdraw()` in the Rust SDK is `transfer()`.](https://fuellabs.github.io/fuels-rs/v0.27.0/wallets/transferring-assets.html?highlight=transfer#transferring-assets)
> 
> This new feature is not about transferring coins within the Fuel chain. It is for withdrawing the base asset (ETH) back to Ethereum. This has not yet been implemented in the rust SDK. It's popping up in the typescript SDK first since that's what's being used for the full Ethereum+Fuel end-to-end tests

Oh, I see; it makes sense not to rename it to `transfer()`. But, `withdraw()` on its own doesn't tell me the wallet is withdrawing to the base chain. Maybe the function name could be more explicit about it?

**Comment by pixelcircuits** [comment]
I'm open to suggestions for a different name other than `withdraw`. What does everyone else think? I used the term `withdraw` because I wanted to avoid using the term `ETH` or `L1`. I guess in the modular stack thesis the correct nomenclature would be `settlement layer`. So maybe something like `withdrawToSettlementLayer`? or `withdrawToBaseLayer`? @adlerjohn @Voxelot

**Comment by digorithm** [comment]
> I'm open to suggestions for a different name other than `withdraw`. What does everyone else think? I used the term `withdraw` because I wanted to avoid using the term `ETH` or `L1`. I guess in the modular stack thesis the correct nomenclature would be `settlement layer`. So maybe something like `withdrawToSettlementLayer`? or `withdrawToBaseLayer`? @adlerjohn @Voxelot

Yeah, we shouldn't be mentioning ETH because the base layer is "pluggable". I like both `withdrawToSettlementLayer` and `withdrawToBaseLayer` 🤔 Maybe `BaseLayer` is more generic/broad here and thus would be better?

**Comment by pixelcircuits** [comment]
ok, renamed `withdraw` to `withdrawToBaseLayer` to make things more clear.

**Review by digorithm — APPROVED** [review]
LGTM -- great stuff!

**Comment by adlerjohn** [comment]
any of the above suggestions seem reasonable to me

File: packages/transactions/src/coders/receipt.ts [hunk 2]
```diff
@@ -682,14 +682,22 @@ export class ReceiptMessageOutCoder extends Coder<ReceiptMessageOut, ReceiptMess
     value.data.forEach((d) => dataInNumbers.push(d));
     const parts: Uint8Array[] = [];
 
+    // encode data padded to nearest word (8 bytes)
```
Comment (luizstacio) [inline packages/transactions/src/coders/receipt.ts:685]: I think here you can use the `ByteArrayCoder `. Like this? https://github.com/FuelLabs/fuels-ts/blob/db8cc6b49616199368463ecd69aae6b3ca0b65d0/packages/transactions/src/coders/input.ts#L73

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/transactions/src/coders/receipt.ts (hunk 2 above)
Comment (pixelcircuits) [reply packages/transactions/src/coders/receipt.ts]: Yep, thanks!

**Review by luizstacio — APPROVED** [review]

**Pull request merged**